apps/container/src/views/HistoryView.vue

```diff
@@ -246,19 +266,48 @@
 }
 </style>
 <script setup lang="ts">
+import { useQuery } from '@tanstack/vue-query';
+import { conceptsColor } from 'consts';
+import { Enrollment, Enrollments, Users } from 'services';
+import { Concept } from 'types';
+import { computed, ref } from 'vue';
+import ReviewDialog from '@/components/ReviewDialog.vue';
+import CenteredLoading from '@/components/CenteredLoading.vue';
 import PaperCard from '@/components/PaperCard.vue';
 import TableComponent from '@/components/TableComponent.vue';
-import {
-  Users,
-  Enrollments,
-  Enrollment as EnrollmentType,
-  Concept,
-} from 'services';
-import { computed } from 'vue';
-import CenteredLoading from '@/components/CenteredLoading.vue';
 import FeedbackAlert from '@/components/FeedbackAlert.vue';
-import { ref } from 'vue';
-import { useQuery } from '@tanstack/vue-query';
+import { checkEAD } from 'utils';
+
+const showDialog = ref(false);
+const selectedEnrollment = ref<Enrollment | undefined>();
+
+const tags = ref<string[]>([]);
+const handleOpenDialog = (
+  enrollment: Enrollment,
+  type: 'teoria' | 'pratica',
+) => {
+  const processedEnrollment: Enrollment = {
+    ...enrollment,
+    teoria: type !== 'pratica' ? enrollment.teoria : null,
+    pratica: type === 'pratica' ? enrollment.pratica : null,
+  };
+  selectedEnrollment.value = processedEnrollment;
+
+  const isEAD = checkEAD(enrollment.year || 0, enrollment.quad || 0);
```

Comment:
Prefiro não colocar esses fallbacks que não fazem muito sentido
```suggestion
  const isEAD = enrollment.year && enrollment.quad  &&  checkEAD(enrollment.year, enrollment.quad);
```